### PR TITLE
chore: Adjust server-less tooling for the new AWS Lambda "provided" run-times

### PR DESCRIPTION
**Pull request by zatlodan**
**What this PR does**:

The "old" `go1.x` runtime is being deprecated.

Changes the CMD server-less build tooling to be compatible with the "new" AWS lambda run-times **by default**.

The "new" GoLang compatible AWS Lambda run-times expect an executable with the name `bootstrap` to be provided.
The `handler` configuration is ignored when using the "new" runtime.

See https://aws.amazon.com/blogs/compute/migrating-aws-lambda-functions-from-the-go1-x-runtime-to-the-custom-runtime-on-amazon-linux-2/ for more info.

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by mapno — COMMENTED** [review]
LGMT. Can you add a changelog entry specifying what's being deprecated? Thanks!

**Comment by zatlodan** [comment]
> LGMT. Can you add a changelog entry specifying what's being deprecated? Thanks!

Thank you, and sorry for the late response.

I've added a changelog entry. 
Since this change only affects the build tooling, I didn't mark it as a breaking change, 
although it might require adjustments for users relying on the previous build process.

Please let me know if any further adjustments are needed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## main / unreleased
 
+
+* [CHANGE] Change the AWS Lambda serverless build tooling output from "main" to "bootstrap" [#3852](https://github.com/grafana/tempo/pull/3852) (@zatlodan)
```
Comment (mapno) [inline CHANGELOG.md:4]:
```suggestion
* [CHANGE] **BREAKING CHANGE** Change the AWS Lambda serverless build tooling output from "main" to "bootstrap". Refer to https://aws.amazon.com/blogs/compute/migrating-aws-lambda-functions-from-the-go1-x-runtime-to-the-custom-runtime-on-amazon-linux-2/ for migration steps [#3852](https://github.com/grafana/tempo/pull/3852) (@zatlodan)
```

**Review by mapno — COMMENTED** [review]
I'd like to be a bit more explicit with the migration.

**Comment by zatlodan** [comment]
> I'd like to be a bit more explicit with the migration.

I agree, it is better to be more explicit here.
I was initially not sure if I should mark it as a breaking change, as I am not used to writing change-logs.

**Comment by zatlodan** [comment]
@mapno Can we get an update on this?

**Review by mapno — APPROVED** [review]

**Comment by mapno** [comment]
LGTM, thanks for the contribution

**Pull request merged**